visualization/app/codeCharta/util/fileDownloader.ts

```diff
@@ -1,27 +1,44 @@
 import angular from "angular"
 import * as d3 from "d3"
-import { CCFile, CodeMapNode } from "../codeCharta.model"
+import { CCFile, CodeMapNode, BlacklistType, BlacklistItem, FileSettings } from "../codeCharta.model"
+import { DownloadCheckboxNames } from "../ui/dialog/dialog.download"
 
 export class FileDownloader {
-	private static CC_FILE_EXTENSION = ".cc.json"
-
-	public static downloadCurrentMap(file: CCFile) {
-		const data = this.getProjectDataAsCCJsonFormat(file)
-		this.downloadData(data, this.getNewFileName(file))
+	public static downloadCurrentMap(file: CCFile, downloadSettingsNames: string[], fileName: string) {
+		const data = this.getProjectDataAsCCJsonFormat(file, downloadSettingsNames)
+		this.downloadData(data, fileName)
 	}
 
-	private static getProjectDataAsCCJsonFormat(file: CCFile) {
-		let newFileName = this.getNewFileName(file)
+	private static getProjectDataAsCCJsonFormat(file: CCFile, downloadSettingsNames: string[]) {
+		const s: FileSettings = file.settings.fileSettings
 
-		return {
-			fileName: newFileName,
+		let downloadObject: any = {
```

Comment:
Don't we have an Interface for our cc.json?

Reply:
I just created an ExportCCFile interface